tests/test_lifeforms.py

```diff
@@ -51,3 +52,77 @@ def test_custom_validate_input_shape():
     """Test if error is raised when input array has wrong dimensions"""
     with pytest.raises(ValueError):
         sg.lifeforms.Custom(np.random.choice([0, 1], size=(1, 3, 3)))
+
+
+def test_parse_plaintext_layout():
+    """Test if parse_plaintext_layout parses sample text correctly"""
+    layout = parse_plaintext_layout(
+        """!Name: Name
+! comment
+..O
+.O
+O
+"""
+    )
+    assert len(layout.shape) == 2
+    assert layout.shape == (3, 3)
+    assert np.sum(layout - [[0, 0, 1], [0, 1, 0], [1, 0, 0]]) == 0
+
+def test_parse_plaintext_layout_list():
+    """Test if parse_plaintext_layout parses sample list input correctly"""
+    layout = parse_plaintext_layout(['..O','.O','O'])
+    assert len(layout.shape) == 2
+    assert layout.shape == (3, 3)
+    assert np.sum(layout - [[0, 0, 1], [0, 1, 0], [1, 0, 0]]) == 0
+
+def test_parse_plaintext_layout_letters_error():
+    """Test if parse_plaintext_layout checks proper letter in input"""
+    with pytest.raises(ValueError):
+        parse_plaintext_layout(['..O','.OX','O'])
+
+@pytest.mark.skip
+def test_glider_lifeform(lifeform):
+    """Test if lifeform is a proper Glider"""
+    assert len(lifeform.size) == 2
+    assert lifeform.size == (3, 3)
+    assert np.sum(lifeform.layout - [[0, 1, 0], [0, 0, 1], [1, 1, 1]]) == 0
+    assert lifeform.name == "Glider"
+    assert lifeform.author == "Richard K. Guy"
+
+
+def test_lifeform_parse_cells():
+    """Test if lifeform is properly parsed from text"""
+    lifeform = parse_cells(
+        """!Name: Glider
+!Author: Richard K. Guy
+!The smallest, most common, and first discovered spaceship.
+!www.conwaylife.com/wiki/index.php?title=Glider
+.O
+..O
+OOO"""
+    )
+    test_glider_lifeform(lifeform)
+
+
+def test_lifeform_parse_cells_file(tmp_path):
```

Comment:
Hi @eugenpt , the CI/CD error is coming form the missing tmp_path fixture. I'm not sure if you can see [this report](https://dev.azure.com/ljvmiranda/ljvmiranda/_build/results?buildId=592&view=logs&jobId=9af4ab1a-d688-5af2-ddff-801f42c95c19&j=9af4ab1a-d688-5af2-ddff-801f42c95c19&t=96e3a3e6-3ea4-5276-659b-298ea9f50550 ), but I'll just put it here for good measure:

```
2020-05-28T06:08:41.8312252Z ##[section]Starting: Run tests with coverage
2020-05-28T06:08:41.8322706Z ==============================================================================
2020-05-28T06:08:41.8323317Z Task         : Command line
2020-05-28T06:08:41.8323799Z Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
2020-05-28T06:08:41.8324234Z Version      : 2.164.1
2020-05-28T06:08:41.8324595Z Author       : Microsoft Corporation
2020-05-28T06:08:41.8325077Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
2020-05-28T06:08:41.8325588Z ==============================================================================
2020-05-28T06:08:41.9610709Z Generating script.
2020-05-28T06:08:41.9626404Z ========================== Starting Command Output ===========================
2020-05-28T06:08:41.9650307Z [command]/bin/bash --noprofile --norc /home/vsts/work/_temp/ce6d4c44-20fe-49e9-b351-debdba5406bd.sh
2020-05-28T06:08:42.5801986Z Requirement already satisfied: pytest in /opt/hostedtoolcache/Python/3.6.10/x64/lib/python3.6/site-packages (3.6.4)
2020-05-28T06:08:42.5901647Z Requirement already satisfied: pytest-cov in /opt/hostedtoolcache/Python/3.6.10/x64/lib/python3.6/site-packages (2.9.0)
2020-05-28T06:08:42.6159506Z Requirement already satisfied: attrs>=17.4.0 in /opt/hostedtoolcache/Python/3.6.10/x64/lib/python3.6/site-packages (from pytest) (19.3.0)
2020-05-28T06:08:42.6417926Z Requirement already satisfied: six>=1.10.0 in /opt/hostedtoolcache/Python/3.6.10/x64/lib/python3.6/site-packages (from pytest) (1.15.0)
2020-05-28T06:08:42.6444309Z Requirement already satisfied: more-itertools>=4.0.0 in /opt/hostedtoolcache/Python/3.6.10/x64/lib/python3.6/site-packages (from pytest) (8.3.0)
2020-05-28T06:08:42.6480426Z Requirement already satisfied: pluggy<0.8,>=0.5 in /opt/hostedtoolcache/Python/3.6.10/x64/lib/python3.6/site-packages (from pytest) (0.7.1)
2020-05-28T06:08:42.6504531Z Requirement already satisfied: atomicwrites>=1.0 in /opt/hostedtoolcache/Python/3.6.10/x64/lib/python3.6/site-packages (from pytest) (1.4.0)
2020-05-28T06:08:42.6522150Z Requirement already satisfied: setuptools in /opt/hostedtoolcache/Python/3.6.10/x64/lib/python3.6/site-packages (from pytest) (46.4.0)
2020-05-28T06:08:42.6781846Z Requirement already satisfied: py>=1.5.0 in /opt/hostedtoolcache/Python/3.6.10/x64/lib/python3.6/site-packages (from pytest) (1.8.1)
2020-05-28T06:08:42.6806822Z Requirement already satisfied: coverage>=4.4 in /opt/hostedtoolcache/Python/3.6.10/x64/lib/python3.6/site-packages (from pytest-cov) (5.1)
2020-05-28T06:08:44.6236335Z ============================= test session starts ==============================
2020-05-28T06:08:44.6240122Z platform linux -- Python 3.6.10, pytest-3.6.4, py-1.8.1, pluggy-0.7.1
2020-05-28T06:08:44.6241195Z rootdir: /home/vsts/work/1/s, inifile:
2020-05-28T06:08:44.6242142Z plugins: cov-2.9.0
2020-05-28T06:08:49.3952736Z collected 72 items
2020-05-28T06:08:49.3967766Z 
2020-05-28T06:08:49.4281176Z tests/test_board.py ....                                                 [  5%]
2020-05-28T06:08:50.6606478Z tests/test_lifeforms.py .........................................s.E.    [ 68%]
2020-05-28T06:08:50.6916278Z tests/test_rules.py ................                                     [ 90%]
2020-05-28T06:08:51.0837475Z tests/test_simulator.py .......                                          [100%]
2020-05-28T06:08:51.0840894Z 
2020-05-28T06:08:51.0841802Z ==================================== ERRORS ====================================
2020-05-28T06:08:51.0842652Z _______________ ERROR at setup of test_lifeform_parse_cells_file _______________
2020-05-28T06:08:51.0843433Z file /home/vsts/work/1/s/tests/test_lifeforms.py, line 107
2020-05-28T06:08:51.0844204Z   def test_lifeform_parse_cells_file(tmp_path):
2020-05-28T06:08:51.0845574Z E       fixture 'tmp_path' not found
2020-05-28T06:08:51.0846709Z >       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, monkeypatch, no_cover, pytestconfig, record_property, record_xml_attribute, record_xml_property, recwarn, tmpdir, tmpdir_factory
2020-05-28T06:08:51.0848754Z >       use 'pytest --fixtures [testpath]' for help on them.
2020-05-28T06:08:51.0849376Z 
2020-05-28T06:08:51.0851746Z /home/vsts/work/1/s/tests/test_lifeforms.py:107
2020-05-28T06:08:51.0853022Z -------- generated xml file: /home/vsts/work/1/s/junit/test-results.xml --------
2020-05-28T06:08:51.0854490Z 
2020-05-28T06:08:51.0856580Z ---------- coverage: platform linux, python 3.6.10-final-0 -----------
2020-05-28T06:08:51.0857203Z Coverage HTML written to dir htmlcov
2020-05-28T06:08:51.0857730Z Coverage XML written to file coverage.xml
2020-05-28T06:08:51.0858259Z 
2020-05-28T06:08:51.0858720Z ================ 70 passed, 1 skipped, 1 error in 6.46 seconds =================
2020-05-28T06:08:51.2850351Z 
2020-05-28T06:08:51.2924702Z ##[error]Bash exited with code '1'.
2020-05-28T06:08:51.2942331Z ##[section]Finishing: Run tests with coverage
```

Reply:
seems like pytest version conflict.. will try different temp files interface